sdl-debezium/src/main/scala/io/smartdatalake/workflow/dataobject/DebeziumCdcDataObject.scala

```diff
@@ -0,0 +1,329 @@
+    }
+
+    val defaultOffsetProperties: Map[String, String] = Map(
+      "offset.storage" -> "io.smartdatalake.debezium.SDLBDebeziumOffsetStorage",
+      "offset.storage.sdlb.data.object.id" -> this.id.id,
+      "offset.flush.interval.ms" -> "10000")
+
+    // If duplicate offset properties are set, prefer the ones the user has set in the config file
+    props = props ++ defaultOffsetProperties.map {
+      case (key, value) => if (props.contains(key)) key -> props(key) else key -> defaultOffsetProperties(key)
+    }
+
+    val defaultProperties: Map[String, String] = Map(
+      "topic.prefix" -> table.fullName,
+      "include.schema.changes" -> "false",
+      "name" -> id.toString,
+      "tombstones.on.delete" -> "false",
+      "decimal.handling.mode" -> "string" // string, because double cannot handle fully completely decimal value and precise is not possible because spark returns an error when processing -> Root cause is 'ClassCastException: class java.math.BigDecimal cannot be cast to class [B (java.math.BigDecimal and [B are in module java.base of loader 'bootstrap')' correct data type can be adjusted f.ex. with a cast in a transformer
+    )
+
+    // If duplicate default properties are set, prefer the ones the user has set in the config file
+    props = props ++ defaultProperties.map {
+      case (key, value) => if (props.contains(key)) key -> props(key) else key -> defaultProperties(key)
+    }
+
+    // Always overwrite table.include.list property to include only the changes of the table specified in the data object
+    props = props ++ Map("table.include.list" -> table.fullName)
+
+    val propsForEngine = new Properties();
+    props.foreach { case (key, value) => propsForEngine.setProperty(key, value) }
+    propsForEngine
+  }
+
+  override def factory: FromConfigFactory[DataObject] = DebeziumCdcDataObject
+
+  override def getSparkDataFrame(partitionValues: Seq[PartitionValues])(implicit context: ActionPipelineContext): DataFrame = {
+
+    val spark = context.sparkSession
+
+    def getRecordsFromDebeziumEngine(
+                           properties: Properties,
+                           changeConsumer: DebeziumEngine.ChangeConsumer[ChangeEvent[SourceRecord, SourceRecord]] with HasRecords[SourceRecord],
+                           executorService: ExecutorService = Executors.newSingleThreadExecutor,
+                           timeoutMilliSeconds: Int = 10000
+                         ): Seq[SourceRecord] = {
+
+      val completionCallback = new DebeziumCompletionCallback(executorService)
+
+      val engine = DebeziumEngine.create(classOf[Connect])
+        .using(properties)
+        .notifying(changeConsumer)
+        .using(completionCallback)
+        .build()
+
+      executorService.execute(engine)
+
+      Thread.sleep(timeoutMilliSeconds)
```

Comment:
this is the first thing I find, after some searching, that looks not so clean. Why do you need to launch a thread in the background and then wait 10 seconds but still return the records without ever waiting for the Thread to complete? I'm wondering how can you know when "everything" has been fetched?